packages/core/component-with-state.ts

```diff
@@ -0,0 +1,63 @@
+import { props } from 'ramda'
+
+import { Component } from './component'
+
+/** @mark draft of types */
+export type StoreDataType = { [key: string]: any }
```

Comment:
Redux store interface is ideal here 
`
{
getState(void): State
subscribe(cb: State => void): { unsubscribe: Function }
dispatch(action: Action): void
}
`

Reply:
Event in case when it won't be redux?

Reply:
yes - interface is super simple, adapter could be written for any implementation

Reply:
as we discussed - we must declare interface, nothing more, but well designed interface

Reply:
ok, I'll try to investigate `mobx` like the second source, just to be sure

Reply:
good idea

Reply:
also tests could be added to have it documented